### PR TITLE
Simplify worker options configuration value propagation

### PR DESCRIPTION
**Pull request by shijiesheng**
**What changed?**

use embedding to simplify assignment

**Why?**

There are duplicate fields in `workerExecutionParameters` and `WorkerOptions` that unnecessarily complicates the logic. 

**How did you test it?**

**Potential risks**

### CODE REVIEW

File: internal/internal_task_handlers.go [hunk 1]
```diff
@@ -396,7 +396,7 @@ func newWorkflowTaskHandler(
 		dataConverter:                  params.DataConverter,
 		contextPropagators:             params.ContextPropagators,
 		tracer:                         params.Tracer,
-		workflowInterceptors:           params.WorkflowInterceptors,
+		workflowInterceptors:           params.WorkflowInterceptorChainFactories,
```
Comment (davidporter-id-au) [inline internal/internal_task_handlers.go:399]: what is a Chain factory?

Comment (shijiesheng) [reply internal/internal_task_handlers.go:399]: It's an option field related to replay. Here is the link. Here it looks like we need a consolidation on the naming. 
https://sourcegraph.com/github.com/uber-go/cadence-client@master/-/blob/internal/worker.go?L200

Comment (shijiesheng) [reply internal/internal_task_handlers.go:399]: Unfortunately, both WorkerOptions and ReplayOptions are exported. But they have different names. 
https://sourcegraph.com/github.com/shijiesheng/cadence-client@master/-/blob/worker/worker.go#L191:43
https://sourcegraph.com/github.com/shijiesheng/cadence-client@master/-/blob/worker/worker.go#L178:37

Comment (shijiesheng) [inline internal/internal_task_handlers.go:399]: I'll rename this as well

**Review by davidporter-id-au — APPROVED** [review]

**Pull request merged**